packages/dom/src/focusable.js

```diff
@@ -77,7 +92,7 @@ export function find( context ) {
 	const elements = context.querySelectorAll( SELECTOR );
 
 	return Array.from( elements ).filter( ( element ) => {
-		if ( ! isVisible( element ) ) {
+		if ( ! isVisible( element ) || skipFocus( element ) ) {
```

Comment:
Should the description of the function be updated to reflect changes applied? Does it make sense to include an entry in the CHANGELOG for the package? Is it a bug fix?

Is it possible to cover it with a unit test to prevent further regressions?

Reply:
I missed this comment. Maybe we can create a follow-up PR?

Reply:
@gziolo It wasn't exactly a bug, more like making `DropdownMenu`'s arrow navigation work properly by adding the `BlockStyles`, which had nested elements. That was causing problems with the declared `Selectors` here: https://github.com/ntsekouras/gutenberg/blob/2a2b2c8ce042bc06c220eab6891b0d477b6d9740/packages/dom/src/focusable.js#L20 . Nested elements were also selected as `focusables` that caused breaking the navigation.

My first thought was introducing [a new attribute in this commit](https://github.com/WordPress/gutenberg/pull/23028/commits/189b35a7e1751ce0235d4cb56a097ac7b6d3bcee#diff-5f5842dcdf7edec05431ef2f0a79b555R70) ( `focusable.js changes` ), but suggested to simplify it..

I would really appreciate any feedback for how to handle this better..

Reply:
`focusable` is the public API on the `@wordpress/dom` package: https://www.npmjs.com/package/@wordpress/dom. It has 15k downloads per week. It's used by 3rd party projects:
<img width="780" alt="Screen Shot 2020-06-23 at 12 55 49" src="https://user-images.githubusercontent.com/699132/85395734-e1e34300-b550-11ea-9ae7-8022bebf8529.png">

Using `tabindex="-1"` – it's a well-established pattern used with keyboard navigation:
https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex

> A negative value (usually tabindex="-1") means that the element is not reachable via sequential keyboard navigation, but could be focused with Javascript or visually by clicking with the mouse. It's mostly useful to create accessible widgets with JavaScript.

It feels like it was missed in the initial implementation and it would mean a bug fix. @ellatrix, can you confirm?

Reply:
It was added by Andrew in #2323. Not sure if iframes were meant to be excluded. I don't really understand the change here. Isn't everything that is tab index -1 already excluded? And an iframe's children can't be queried, so they are also excluded?

Reply:
I think that you need to explicitly mark all children at every nesting level to make it work. It's meant to be addressed with [inert attribute](https://github.com/WICG/inert) but it isn't implemented in all browsers yet.